dadi/lib/storage/http.js

```diff
@@ -54,79 +54,75 @@ HTTPStorage.prototype.get = function ({
         'User-Agent': 'DADI CDN'
       }
     }, res => {
-      let statusCode = res.statusCode
+      let buffers = []
 
-      if (statusCode === 200) {
-        let buffers = []
+      res.on('data', chunk => {
+        buffers.push(chunk)
+      })
 
-        res.on('data', chunk => {
-          buffers.push(chunk)
-        })
+      res.on('end', () => {
+        let statusCode = res.statusCode
 
-        res.on('end', () => {
+        if (statusCode === 200) {
           return resolve(streamifier.createReadStream(Buffer.concat(buffers)))
-        })
-      } else {
-        if (
-          [301, 302, 307].includes(statusCode) &&
-          typeof res.headers.location === 'string'
-        ) {
-          let parsedRedirectUrl = url.parse(res.headers.location)
-
-          parsedRedirectUrl.host = parsedRedirectUrl.host || parsedUrl.host
-          parsedRedirectUrl.port = parsedRedirectUrl.port || parsedUrl.port
-          parsedRedirectUrl.protocol = parsedRedirectUrl.protocol || parsedUrl.protocol
-
-          if (redirects < config.get('http.followRedirects', this.domain)) {
-            return resolve(
-              this.get({
-                redirects: redirects + 1,
-                requestUrl: url.format(parsedRedirectUrl)
-              })
-            )
+        } else {
```

Comment:
This `else` isn't required, because the alternative is the `return` statement on L67. Getting rid of it would reduce the indentation level of L68-L124, which would help in terms of readibility.